File: GamingRoomWeb/src/Entity/Membre.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Membre
 *
 * @ORM\Table(name="membre", uniqueConstraints={@ORM\UniqueConstraint(name="email", columns={"email"})})
 * @ORM\Entity
 * @ORM\Entity(repositoryClass="App\Repository\MembreRepository")
 */
class Membre
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="nom", type="string", length=20, nullable=false)
     */
    private $nom;

    /**
     * @var string
     *
     * @ORM\Column(name="prenom", type="string", length=20, nullable=false)
     */
    private $prenom;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_naissance", type="date", nullable=false)
     */
    private $dateNaissance;

    /**
     * @var string
     *
     * @ORM\Column(name="genre", type="string", length=0, nullable=false)
     */
    private $genre;

    /**
     * @var string
     *
     * @ORM\Column(name="numero_tel", type="string", length=8, nullable=false)
     */
    private $numeroTel;

    /**
     * @var string
     *
     * @ORM\Column(name="email", type="string", length=50, nullable=false)
     */
    private $email;

    /**
     * @var string
     *
     * @ORM\Column(name="password", type="string", length=255, nullable=false)
     */
    private $password;

    /**
     * @var string
     *
     * @ORM\Column(name="image", type="string", length=50, nullable=false)
     */
    private $image;

    /**
     * @var string
     *
     * @ORM\Column(name="role", type="string", length=0, nullable=false)
     */
    private $role;

    /**
     * @var int
     *
     * @ORM\Column(name="point", type="integer", nullable=false)
     */
    private $point = '0';

    /**
     * @var string|null
     *
     * @ORM\Column(name="description", type="string", length=255, nullable=true, options={"default"="NULL"})
     */
    private $description = 'NULL';

    /**
     * @var bool
     *
     * @ORM\Column(name="active", type="boolean", nullable=false)
     */
    private $active;

    /**
     * @var int|null
     *
     * @ORM\Column(name="ban_duration", type="integer", nullable=true, options={"default"="NULL"})
     */
    private $banDuration = NULL;

    /**
     * @var \DateTime|null
     *
     * @ORM\Column(name="last_timeban", type="datetime", nullable=true, options={"default"="NULL"})
     */
    private $lastTimeban = 'NULL';

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrenom(): ?string
    {
        return $this->prenom;
    }

    public function setPrenom(string $prenom): self
    {
        $this->prenom = $prenom;

        return $this;
    }

    public function getDateNaissance(): ?\DateTimeInterface
    {
        return $this->dateNaissance;
    }

    public function setDateNaissance(\DateTimeInterface $dateNaissance): self
    {
        $this->dateNaissance = $dateNaissance;

        return $this;
    }

    public function getGenre(): ?string
    {
        return $this->genre;
    }

    public function setGenre(string $genre): self
    {
        $this->genre = $genre;

        return $this;
    }

    public function getNumeroTel(): ?string
    {
        return $this->numeroTel;
    }

    public function setNumeroTel(string $numeroTel): self
    {
        $this->numeroTel = $numeroTel;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): self
    {
        $this->email = $email;

        return $this;
    }

    public function getPassword(): ?string
    {
        return $this->password;
    }

    public function setPassword(string $password): self
    {
        $this->password = $password;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getRole(): ?string
    {
        return $this->role;
    }

    public function setRole(string $role): self
    {
        $this->role = $role;

        return $this;
    }

    public function getPoint(): ?int
    {
        return $this->point;
    }

    public function setPoint(int $point): self
    {
        $this->point = $point;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getActive(): ?bool
    {
        return $this->active;
    }

    public function setActive(bool $active): self
    {
        $this->active = $active;

        return $this;
    }

    public function getBanDuration(): ?int
    {
        return $this->banDuration;
    }

    public function setBanDuration(?int $banDuration): self
    {
        $this->banDuration = $banDuration;

        return $this;
    }

    public function getLastTimeban(): ?\DateTimeInterface
    {
        return $this->lastTimeban;
    }

    public function setLastTimeban(?\DateTimeInterface $lastTimeban): self
    {
        $this->lastTimeban = $lastTimeban;

        return $this;
    }


}

File: GamingRoomMoblie/src/com/esprit/gamingroom/entities/Membre.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package com.esprit.gamingroom.entities;

import java.sql.Date;
import java.sql.Timestamp;

/**
 *
 * @author Sonia
 */
public class Membre {

   
    public enum Role{
        Admin,
        Membre,
        Coach 
    }
    public enum Genre{
        Femme,
        Homme
    }
    private int id;
    private String nom;
    private String prenom;
    private Date date_naissance;
    private Genre genre;
    private String tel;
    private String email;
    private String password;
    private String image;
    private Role role;
    private int point ;
    private String description;
    private boolean active;
    private int ban_duration;
    private Timestamp last_timeban;

    public Membre() {
    }

    public Membre(int id) {
        this.id = id;
    }

    public Membre(int id, String nom, String prenom, Date date_naissance, Genre genre, String tel, String email, String image) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.date_naissance = date_naissance;
        this.genre = genre;
        this.tel = tel;
        this.email = email;
        this.image = image;
    }
    
    

    public Membre(int id, int point, boolean active, int ban_duration, Timestamp last_timeban) {
        this.id = id;
        this.point = point;
        this.active = active;
        this.ban_duration = ban_duration;
        this.last_timeban = last_timeban;
    }

   
    

    public Membre(int id, boolean active) {
        this.id = id;
        this.active = active;
    }
    

    public Membre(int id, int point) {
        this.id = id;
        this.point = point;
    }
    

    public Membre(int id, String nom, String prenom, Date date_naissance, Genre genre, String tel, String email, String password, String image, Role role, int point, String description, boolean active, int ban_duration, Timestamp last_timeban) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.date_naissance = date_naissance;
        this.genre = genre;
        this.tel = tel;
        this.email = email;
        this.password = password;
        this.image = image;
        this.role = role;
        this.point = point;
        this.description = description;
        this.active = active;
        this.ban_duration = ban_duration;
        this.last_timeban = last_timeban;
    }

    public Membre(String nom, String prenom, Date date_naissance, Genre genre, String tel, String email, String password, String image, Role role, int point, String description, boolean active, int ban_duration, Timestamp last_timeban) {
        this.nom = nom;
        this.prenom = prenom;
        this.date_naissance = date_naissance;
        this.genre = genre;
        this.tel = tel;
        this.email = email;
        this.password = password;
        this.image = image;
        this.role = role;
        this.point = point;
        this.description = description;
        this.active = active;
        this.ban_duration = ban_duration;
        this.last_timeban = last_timeban;
    }

    public Membre(int id, String nom, String prenom, Date date_naissance, Genre genre, String tel, String email, String password, Role role, boolean active) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.date_naissance = date_naissance;
        this.genre = genre;
        this.tel = tel;
        this.email = email;
        this.password = password;
        this.role = role;
        this.active = active;
    }

    public Membre(int id, String nom, String prenom, Date date_naissance, Genre genre, String tel, String email, String password, String image, Role role, boolean active) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.date_naissance = date_naissance;
        this.genre = genre;
        this.tel = tel;
        this.email = email;
        this.password = password;
        this.image = image;
        this.role = role;
        this.active = active;
    }

  
    

    public Membre(String nom, String prenom, Date date_naissance, Genre genre, String tel, String email, String password, String image, Role role, boolean active) {
        this.nom = nom;
        this.prenom = prenom;
        this.date_naissance = date_naissance;
        this.genre = genre;
        this.tel = tel;
        this.email = email;
        this.password = password;
        this.image = image;
        this.role = role;
        this.active = active;
    }

    public Membre(int id, String nom, String prenom, Date date_naissance, Genre genre, String tel, String email, String password, String image, Role role, String description, boolean active) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.date_naissance = date_naissance;
        this.genre = genre;
        this.tel = tel;
        this.email = email;
        this.password = password;
        this.image = image;
        this.role = role;
        this.description = description;
        this.active = active;
    }

    public Membre(int id, String nom, String prenom, Date date_naissance, Genre genre, String tel, String email, String password, Role role, String description, boolean active) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.date_naissance = date_naissance;
        this.genre = genre;
        this.tel = tel;
        this.email = email;
        this.password = password;
        this.role = role;
        this.description = description;
        this.active = active;
    }
    
    
    public Membre(String nom, String prenom, Date date_naissance, Genre genre, String tel, String email, String password, String image, Role role, String description, boolean active) {
        this.nom = nom;
        this.prenom = prenom;
        this.date_naissance = date_naissance;
        this.genre = genre;
        this.tel = tel;
        this.email = email;
        this.password = password;
        this.image = image;
        this.role = role;
        this.description = description;
        this.active = active;
    }
    
    

   

    public int getId() {
        return id;
    }

    public String getNom() {
        return nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public Date getDate_naissance() {
        return date_naissance;
    }

    public Genre getGenre() {
        return genre;
    }

    public String getTel() {
        return tel;
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }

    public String getImage() {
        return image;
    }

    public Role getRole() {
        return role;
    }

    public int getPoint() {
        return point;
    }


    public String getDescription() {
        return description;
    }

    public boolean getActive() {
        return active;
    }

    public int getBan_duration() {
        return ban_duration;
    }

    public Timestamp getLast_timeban() {
        return last_timeban;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public void setDate_naissance(Date date_naissance) {
        this.date_naissance = date_naissance;
    }

    public void setGenre(Genre genre) {
        this.genre = genre;
    }

    public void setTel(String tel) {
        this.tel = tel;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public void setRole(Role role) {
        this.role = role;
    }

    public void setPoint(int point) {
        this.point = point;
    }


    public void setDescription(String description) {
        this.description = description;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    public void setBan_duration(int ban_duration) {
        this.ban_duration = ban_duration;
    }

    public void setLast_timeban(Timestamp last_timeban) {
        this.last_timeban = last_timeban;
    }

    @Override
    public String toString() {
        return "Membre{" + "id=" + id + ", nom=" + nom + ", prenom=" + prenom + ", date_naissance=" + date_naissance + ", genre=" + genre + ", tel=" + tel + ", email=" + email + ", password=" + password + ", image=" + image + ", role=" + role + ", point=" + point +  ", description=" + description + ", active=" + active + ", ban_duration=" + ban_duration + ", last_timeban=" + last_timeban + '}'+"\n";
    }
    
    
    
    
    
}
